lib/hash.js

```diff
@@ -1,8 +1,24 @@
 'use strict'
 
+function getInvalidatingOptions (config) {
+  return [
+    'instrument',
+    'instrumenter',
+    'sourceMap'
+  ].reduce((acc, optName) => {
+    acc[optName] = config[optName]
+    return acc
+  }, {})
+}
+
 module.exports = {
-  salt: JSON.stringify({
-    istanbul: require('istanbul-lib-coverage/package.json').version,
-    nyc: require('../package.json').version
-  })
+  salt (config) {
+    return JSON.stringify(Object.assign(
+      {
+        istanbul: require('istanbul-lib-coverage/package.json').version,
```

Comment:
I think `istanbul-lib-instrument` is needed instead of `istanbul-lib-coverage`.  `istanbul-lib-coverage` is bundled with `nyc` where `istanbul-lib-instrument` is not.  So it's impossible for you to make `nyc@13.1.0` use `istanbul-lib-coverage@2.0.3`.  It is possible to force `nyc@13.1.0` to use `istanbul-lib-instrument@3.0.1`.

I'd prefer you structure the object to be stringified as:
```js
{
  modules: {
   'istanbul-lib-instrument': require(...).version,
   nyc: require(...).version
  },
  nycrc: getInvalidatingOptions(config)
}
```